src/land/imsfv3_scf2ioda.py

```diff
@@ -128,6 +121,30 @@ def _read(self):
         DimDict['Location'] = len(self.outdata[('dateTime', 'MetaData')])
 
 
+def get_observation_time(filename, sncv, ncd):
```

Comment:
@CoryMartin-NOAA and @YoulongXia-NOAA here's the function which will try to get the `time` from the netCDF file

if not present will proceed to get from filename, and I've adapted the wonderful regex to get the last occurrence of 8 consecutive digits.  Agree we would like to not have to do this and it is potentially a misleading or plain not correct.  Only here as a fallback